modules/internal/ChapelHashtable.chpl

```diff
@@ -356,6 +356,121 @@ module ChapelHashtable {
       }
     }
 
+    // #### iterate over filled slots ####
+
+    // The leader chunks up the number of _filled slots_ in the table
+    // into one or more followers of (a..#c), which is a dense range
+    // over filled slots only.
+    // The followers must then iterate through all the slots in the
+    // table until they hit 'a' full slots, and then yield the next
+    // (c - a + 1) full slots.
+    // This should perform in roughly linear time unless the slots
+    // in the table are poorly distributed or there are a lot of
+    // deleted slots in the table.
+    // Performance will also degrade if a different leader is used
+    // that creates a lot of smaller ranges over the iteration
+    // space.
+
+    iter allFilledSlots() ref {
+      for i in 0..#tableSize do
+        if table[i].isFull() then yield table[i];
+    }
+
+    iter allFilledSlots(param tag: iterKind) ref
+      where tag == iterKind.standalone {
```

Comment:
Why not just use  `_allSlots` iterator with a conditional in it for the standalone iterator, like

``` chapel
for slot in _allSlots(tableSize, tag=tag) {
  ref element = table[slot];
  if element.isFull {
    yield element;
  }
}
```

?

Reply:
Or is this needing to match the iteration pattern of the leader/follower in that case? Anyway it seems it could be factored better. 

Name-wise, allSlots returns integers (slot indices). This one could also return integers, or it could have a name indicating it's returning the table entry records.

Reply:
I can remove the serial iterator and just call `allSlots`, as you say.